File: shoring/flywheel/src/main/java/com/panopset/flywheel/LineFeedRules.java
package com.panopset.flywheel;

import com.panopset.compat.Stringop;

public class LineFeedRules {

  private Boolean lineBreaks;
  private Boolean listBreaks;

  public static final LineFeedRules LINE_BREAKS = new LineFeedRules(true, false);
  public static final LineFeedRules LIST_BREAKS = new LineFeedRules(false, true);
  public static final LineFeedRules FULL_BREAKS = new LineFeedRules(true, true);
  public static final LineFeedRules FLATTEN = new LineFeedRules(false, false);
  
  public LineFeedRules() {}

  public LineFeedRules(String lineBreaks, String listBreaks) {
    this.lineBreaks = Stringop.parseBoolean(lineBreaks);
    this.listBreaks = Stringop.parseBoolean(listBreaks);
  }

  public LineFeedRules(Boolean lineBreaks, Boolean listBreaks) {
    this.lineBreaks = lineBreaks;
    this.listBreaks = listBreaks;
  }

  public Boolean getLineBreaks() {
   if (lineBreaks == null) {
    lineBreaks = true;
   }
   return lineBreaks;
  }

  public void setLineBreaks(Boolean lineBreaks) {
   this.lineBreaks = lineBreaks;
  }

  public Boolean getListBreaks() {
   if (listBreaks == null) {
    listBreaks = true;
   }
   return listBreaks;
  }

  public void setListBreaks(Boolean listBreaks) {
   this.listBreaks = listBreaks;
  }
  
  @Override
  public String toString() {
    return String.format("LineB:%s ListB:%s", getLineBreaks(), getListBreaks());
  }
}

File: code/shoring/flywheel/src/main/kotlin/com/panopset/flywheel/LineFeedRules.kt
package com.panopset.flywheel

data class LineFeedRules(var lineBreaks: Boolean, var listBreaks: Boolean) {
    override fun toString(): String {
        return "LineB:$lineBreaks ListB:$listBreaks"
    }
}

const val lineBreaksTrue = true
const val lineBreaksFalse = false
const val listBreaksTrue = true
const val listBreaksFalse = false

val LINE_BREAKS = LineFeedRules(lineBreaksTrue, listBreaksFalse)
val LIST_BREAKS = LineFeedRules(lineBreaksFalse, listBreaksTrue)
val FULL_BREAKS = LineFeedRules(lineBreaksTrue, listBreaksTrue)
val LFR_FLATTEN = LineFeedRules(lineBreaksFalse, listBreaksFalse)
